HealthVault/Classes/Types/Private/MHVThingQueryResultInternal.h

```diff
@@ -54,7 +54,7 @@
 
 @end
 
-@interface MHVThingQueryResultCollectionInternal : MHVCollection<MHVThingQueryResultInternal *>
+@interface MHVThingQueryResultCollectionInternal : MHVCollectionInternal<MHVThingQueryResultInternal *>
```

Comment:
This will need to be refactored too. Though this is an internal class we should delete MHVCollectionInternal.